packages/search-ui-app-search-connector/src/__tests__/AppSearchAPIConnector.test.js

```diff
@@ -561,6 +580,98 @@ describe("AppSearchAPIConnector", () => {
       });
     });
 
+    describe("beforeAutocompleteResultsCall", () => {
+      it("will use the beforeAutocompleteResultsCall parameter to amend option parameters to the search endpoint call", async () => {
+        const state = {
+          current: 2,
+          searchTerm: "searchTerm"
+        };
+
+        const queryConfig = {
+          results: {
+            sortDirection: "desc",
+            sortField: "name",
+            resultsPerPage: 5
+          }
+        };
+
+        const beforeAutocompleteResultsCall = (options, next) => {
+          // Remove sort_direction and sort_field
+          // eslint-disable-next-line no-unused-vars
+          const { sort, ...rest } = options;
+          return next({
+            ...rest,
+            // Add test
+            test: "value"
+          });
+        };
+
+        await subject(state, queryConfig, { beforeAutocompleteResultsCall });
+
+        expect(getLastSearchCall()).toEqual([
+          state.searchTerm,
+          {
+            page: {
+              size: 5
+            },
+            test: "value"
+          }
+        ]);
+      });
+    });
+
+    describe("beforeAutocompleteSuggestionsCall", () => {
+      it("will use the beforeAutocompleteSuggestionsCall parameter to amend option parameters to the search endpoint call", async () => {
+        const state = {
+          current: 2,
+          searchTerm: "searchTerm",
+          sortDirection: "desc",
+          sortField: "name"
+        };
+
+        const queryConfig = {
+          suggestions: {}
+        };
+
+        const beforeAutocompleteSuggestionsCall = (options, next) => {
+          // Remove sort_direction and sort_field
+          // eslint-disable-next-line no-unused-vars
+          const { sort, ...rest } = options;
+          return next({
+            ...rest,
+            // Add test
+            test: "value"
+          });
+        };
+
+        await subject(state, queryConfig, {
+          beforeAutocompleteSuggestionsCall
+        });
+
+        expect(getLastSuggestCall()).toEqual([
+          state.searchTerm,
+          {
+            test: "value"
+          }
+        ]);
+
+        await subject(
+          {},
+          { suggestions: {} },
+          {
+            beforeAutocompleteSuggestionsCall: (queryOptions, next) =>
+              next({
+                ...queryOptions,
+                test: "value"
+              })
+          }
+        );
+        // eslint-disable-next-line no-unused-vars
+        const [_, passedOptions] = getLastSuggestCall();
```

Comment:
could we avoid the eslint-disable here by doing `const passedOptions = getLastSuggestCall()[1]` or will that cause a different eslint objection? (no strong feelings either way, just throwing that out there)

Reply:
Yes, I think that would be better.